onnx/defs/experiments/defs.cc

```diff
@@ -56,14 +57,30 @@ NOTE: Currently, it supports data type of float, int32, int64, and bool.
         "1D tensor containing the desired output shape.  First input must be in "
         "CPU context.",
         AttrType::INT)
-    .Input(0, "input", "Input tensor (optional) to provide shape information.", "T", true)
+    .Input(
+        0,
+        "input",
+        "Input tensor (optional) to provide shape information.",
+        "T1",
+        true)
     .Output(
-            0,
-            "output",
-            "Output tensor of constant values specified by 'value'"
-            "argument and its type is specified by the 'dtype' argument", "T")
-    .TypeConstraint("T", { "tensor(float16)", "tensor(float)", "tensor(double)" },
-        "Constrain input and output types to float tensors.");
+        0,
+        "output",
+        "Output tensor of constant values specified by 'value'"
+        "argument and its type is specified by the 'dtype' argument",
+        "T2")
+    .TypeConstraint(
```

Comment:
This is arguably not quite right, but I don't see an easy fix. Could you add a comment?